contributions/whiten.py

```diff
@@ -0,0 +1,191 @@
+        it is estimated from the time series.
+    kwargs : keywords
+        Additional keyword arguments are passed on to the
+        `pycbc.psd.welch` method.
+
+    Returns
+    -------
+    whitened_data : numpy array
+        The whitened time series.
+
+    """
+    if strain.ndim==1:
+        colored_ts = pycbc.types.TimeSeries(strain, delta_t=delta_t)
+        if psd is None:
+            psd = colored_ts.psd(segment_duration, **kwargs)
+        elif isinstance(psd, np.ndarray):
+            assert psd.ndim==1
+            print('WARNING: Assuming PSD delta_f based on delta_t, length of PSD and EVEN length of original time series. Tread carefully!')
+            assumed_duration = delta_t*(2*len(psd)-2)
+            psd = pycbc.types.FrequencySeries(psd, delta_f=1./assumed_duration)
+        elif isinstance(psd, pycbc.types.FrequencySeries):
+            pass
+        else:
+            raise ValueError("Unknown format of PSD.")
+        psd = interpolate(psd, colored_ts.delta_f)
+        max_filter_len = int(max_filter_duration*colored_ts.sample_rate)
+
+        psd = inverse_spectrum_truncation(psd,
+                    max_filter_len=max_filter_len,
+                    low_frequency_cutoff=low_frequency_cutoff,
+                    trunc_method=trunc_method)
+
+        # inv_psd = np.array([0. if num==0. else 1./num for num in psd])
+        inv_psd = 1./psd
+        white_ts = (colored_ts.to_frequencyseries()*inv_psd**0.5).to_timeseries()
+
+        if remove_corrupted:
+            white_ts = white_ts[max_filter_len//2:(len(colored_ts)-max_filter_len//2)]
+
+        return white_ts.numpy()
+
+    elif strain.ndim==2:
+        psds_1d = None
+        if isinstance(psd, np.ndarray):
+            if psd.ndim==1:
+                psds_1d = [psd for _ in strain]
+        if psds_1d is None:
+            if (psd is None) or isinstance(psd, pycbc.types.FrequencySeries):
+                psds_1d = [psd for _ in strain]
+            else:
+                assert len(psd)==len(strain)
+                psds_1d = psd
+
+        white_segments = [whiten(sd_strain, delta_t=delta_t,
+                segment_duration=segment_duration,
+                max_filter_duration=max_filter_duration,
+                trunc_method=trunc_method, remove_corrupted=remove_corrupted,
+                low_frequency_cutoff=low_frequency_cutoff,
+                psd=psd_1d, **kwargs) for sd_strain, psd_1d in zip(strain, psds_1d)]
+        return np.stack(white_segments, axis=0)
+
+    else:
+        raise ValueError("Strain numpy array dimension must be 1 or 2.")
+
+
+def main():
+    parser = ArgumentParser(description="Strain whitening script to be used for development of submission to the MLGWSC-1 mock data challenge. Written by Ondřej Zelenka.")
+
+    parser.add_argument('--verbose', action='store_true', help="Print update messages.")
+    parser.add_argument('--debug', action='store_true', help="Show debug messages.")
```

Comment:
I'd like to see a `--force` option, to allow users to overwrite data if they wish to.